tools/clang/test/SemaHLSL/attributes/wavesize_range.hlsl

```diff
@@ -0,0 +1,126 @@
+// RUN: %dxc -T lib_6_8 -verify %s
+
+// Check the WaveSize attribute is accepted by work graph nodes
+// and appears in the metadata
+
+struct INPUT_RECORD
+{
+  uint DispatchGrid1 : SV_DispatchGrid;
+  uint2 a;
+};
+
+
+
+[Shader("node")]
+[NodeLaunch("broadcasting")]
+[NumThreads(1,1,1)]
+[NodeMaxDispatchGrid(32,1,1)]
+[WaveSize(4, 16, 32)] /* expected-error{{Preferred WaveSize value 32 must be between 4 and 16}} */
+void node01(DispatchNodeInputRecord<INPUT_RECORD> input) { }
+
+
+[Shader("node")]
+[NodeLaunch("broadcasting")]
+[NumThreads(1,1,1)]
+[NodeMaxDispatchGrid(32,1,1)]
+/* expected-error@+2{{Preferred WaveSize value 32 must be between 16 and 16}} */
+/* expected-error@+1{{Minimum WaveSize value 16 must be less than Maximum WaveSize value 16}} */
```

Comment:
I find this error ("Minimum WaveSize value 16 must be less than Maximum WaveSize value 16") confusing and misleading.  Min==Max is a valid degenerate case.  I opened an [issue](https://github.com/microsoft/hlsl-specs/issues/157) about this in order to resolve a comment in the spec proposal, but it seems to have been ignored.

Reply:
Also: this doesn't check error cases of WaveSize with default preferred (`[WaveSize(min,max)]` form), or passing a `0` value explicitly in different combinations of parameters.

I would prefer we allow and translate degenerate cases into the canonical form.  Pseudo code:
- if max==min, set max=0
- if max==0 && preferred!=0:
  - if preferred==min, set preferred=0, else error.